Update the docker installation example (#3972)
circleci/circleci-docs: jekyll/_cci2/building-docker-images.md

The example below shows how you can build and push a Docker image for our [demo docker project](https://github.com/CircleCI-Public/circleci-demo-docker):

```yaml
version: 2.1
jobs:
  build:
    docker:
      - image: circleci/golang:1.13-alpine   # (1)
    steps:
      - checkout
      # ... steps for building/testing app ...

      - setup_remote_docker:   # (2)
          docker_layer_caching: true # (3)

      # build and push Docker image
      - run: |
          TAG=0.1.$CIRCLE_BUILD_NUM
          docker build -t CircleCI-Public/circleci-demo-docker:$TAG .
          echo $DOCKER_PASS | docker login -u $DOCKER_USER --password-stdin  # (4)
          docker push CircleCI-Public/circleci-demo-docker:$TAG
```

> If the primary container you're using doesn't already have the docker CLI installed, then [you'll need to install it](https://docs.docker.com/install/#supported-platforms) somehow.

Reviewer: ```suggestion
If the primary container you are using doesn't already have the docker CLI installed, then [you will need to install it](https://docs.docker.com/install/#supported-platforms) somehow.
```
